server/rest/src/test/java/org/infinispan/rest/resources/HealthCheckResourceTest.java

```diff
@@ -0,0 +1,128 @@
+package org.infinispan.rest.resources;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.infinispan.client.rest.RestResponseInfo.OK;
+import static org.infinispan.client.rest.RestResponseInfo.SERVICE_UNAVAILABLE;
+import static org.infinispan.commons.test.CommonsTestingUtil.tmpDirectory;
+import static org.infinispan.configuration.cache.CacheMode.DIST_SYNC;
+import static org.infinispan.partitionhandling.PartitionHandling.DENY_READ_WRITES;
+
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import org.infinispan.client.rest.RestResponse;
+import org.infinispan.client.rest.RestServerClient;
+import org.infinispan.client.rest.configuration.Protocol;
+import org.infinispan.commons.util.Util;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.globalstate.ConfigurationStorage;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.security.AuthorizationPermission;
+import org.infinispan.security.Security;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.test.fwk.CleanupAfterMethod;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+
+@CleanupAfterMethod
+@Test(groups = "functional", testName = "rest.ServerHealthCheckResourceTest")
+public class HealthCheckResourceTest extends AbstractRestResourceTest {
+
+   private static final String PERSISTENT_LOCATION = tmpDirectory(HealthCheckResourceTest.class.getName());
+   private static final String TEST_CACHE_NAME = "testing-cache";
+
+   private RestServerClient restServerClient;
+
+   {
+      // Invalid configurations will leave the cluster not ready.
```

Comment:
Hmm this is sounding alarm bells. If we have a cluster with many clients and one client incorrectly defines a cache, does that mean we have taken the cluster down for all clients?

Reply:
I would need to test it, but I believe it would make it not ready :(
We need to verify the component registry for the cache to know whether it is running. When the cache is misconfigured, the component registry is not defined. The same happens when we shut down the cache.

I thought of a way to work around this while typing the message. Going to check if that works and will update.

Reply:
Managed to make it work. If the component registry isn't found, we can retrieve the CF in the cache manager to verify whether the cache doesn't exist, has failed to initialize, or has stopped.